tests/test_eval.cpp

```diff
@@ -96,4 +97,15 @@ TEST_SUBMODULE(eval_, m) {
         auto int_class = py::eval("isinstance(42, int)", global);
         return global;
     });
+
+    m.def("test_eval_closure", [global, copy]() {
```

Comment:
Maybe also add the `// test_eval_closure` comment above. In this case, it's a bit over the top since your function is already called that, but I like the existing convention of indicating the Python-side tests that C++-side bindings belong to.